Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+			SharedDoorControls();
+			m_DoorState = OPENING;
+		}
+		m_ResetDefaultTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::CloseDoor() {
+		if (m_DoorState == STOPPED) {
+			SharedDoorControls();
+			m_DoorState = OPENING;
+			m_ChangedDirectionAfterStop = (m_DoorState == m_DoorStateOnStop) ? false : true;
+		}
+		if (m_DoorState == OPEN || m_DoorState == OPENING) {
+			SharedDoorControls();
+			m_DoorState = CLOSING;
+		}
+		m_ResetDefaultTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::StopDoor() {
+		if (m_DoorState == OPENING || m_DoorState == CLOSING) {
+			if (m_DrawMaterialLayerWhenOpen || m_DrawMaterialLayerWhenClosed) { DrawDoorMaterial(); }
+			m_DoorMoveStopTime = m_DoorMoveTime - m_DoorMoveTimer.GetElapsedSimTimeMS();
+			m_DoorStateOnStop = m_DoorState;
+			m_DoorMoveSound.Stop();
+			m_DoorMoveEndSound.Play(m_Pos);
+			m_DoorState = STOPPED;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::SharedDoorControls() {
+		if (m_DoorState == OPEN || m_DoorState == CLOSED) {
+			m_DoorMoveStartSound.Play(m_Pos);
+			m_DoorMoveTimer.Reset();
+			if (m_DoorMaterialDrawn) { EraseDoorMaterial(); }
+			if (m_Door) { m_Door->DeepCheck(true); }
+		} else if (m_DoorState == OPENING || m_DoorState == CLOSING) {
+			m_DoorMoveSound.Stop();
+			if (!m_ResumeAfterStop) {
+				m_DoorDirectionChangeSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_DoorMoveTime - m_DoorMoveTimer.GetElapsedSimTimeMS());
+			} else {
+				m_DoorMoveStartSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_ChangedDirectionAfterStop ? m_DoorMoveTime - m_DoorMoveStopTime : m_DoorMoveStopTime);
+				m_ChangedDirectionAfterStop = false;
+				m_ResumeAfterStop = false;
+			}
+		} else if (m_DoorState == STOPPED) {
+			if (m_DoorMaterialDrawn) { EraseDoorMaterial(); }
+			m_ResumeAfterStop = true;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::Update() {
+		Actor::Update();
+
+		if (m_Door && m_Door->IsAttached()) {
+			if (m_DoorState != STOPPED && m_SensorTimer.IsPastSimMS(m_SensorInterval)) { UpdateSensors(); }
+			UpdateDoorAttachable();
+		}
+
+		if (m_Door && !ApplyAttachableForces(m_Door)) {
+			// Remove the material representation if the door fell off
+			EraseDoorMaterial();
+			// Start the spinning out of control animation for the motor, start it slow
+			m_SpriteAnimDuration *= 4;
+			m_Door = 0;
+		}
+
+		if (m_SpriteAnimMode == LOOPWHENOPENCLOSE && m_FrameCount > 1 && (m_DoorState == OPENING || m_DoorState == CLOSING) && m_SpriteAnimTimer.IsPastSimMS(m_SpriteAnimDuration)) {
```

Comment:
Would it not be more correct to call the animation mode `LOOPWHENOPENINGCLOSING` or `LOOPWHENOPENINGORCLOSING`?
I guess that makes it even harder to read though.

Reply:
I'm considering changing all enums to CamelCase so they're easier to read.